src/libbson/src/bson/bcon.h

```diff
@@ -167,17 +168,21 @@ typedef enum {
    BCON_TYPE_ERROR,
 } bcon_type_t;
 
+MC_DISABLE_PADDED_WARNING_BEGIN
```

Comment:
I would rather build with `-Wno-padded` than suppress. This seems like a case of "we know this causes a warning, but we choose not to address it".

Suppressing `-Wunsafe-buffer-usage` seems like case of "we know this causes a warning, but this is deliberate" as is the case of `-Wunsafe-buffer-usage`